leaflet-osm.js

```diff
@@ -141,11 +141,14 @@ L.OSM.DataLayer = L.FeatureGroup.extend({
     }
   },
 
-  buildFeatures: function (xml) {
-    var features = L.OSM.getChangesets(xml),
-      nodes = L.OSM.getNodes(xml),
-      ways = L.OSM.getWays(xml, nodes),
-      relations = L.OSM.getRelations(xml, nodes, ways);
+  buildFeatures: function (data) {
+
+    const parser = (data.version) ? L.OSM.JSONParser : L.OSM.XMLParser;
```

Comment:
Presumably this is relying on `data` either being a JSON object (that has a `version` member) or an XML object (that doesn't) but that seems quite fragile... Could we use `instanceof` instead to check the type of the data?

Reply:
Agreed, data.version is quite fragile, and I also don't really like this approach. I need to check for some alternatives.

Reply:
I didn't have much luck with instanceof,  `(data instanceof Document)`  fails in the test environment with `ReferenceError: Document is not defined.`

https://developer.mozilla.org/en-US/docs/Web/API/Document/documentElement might be better then checking data.version?

```
    const parser = (data.documentElement ? L.OSM.XMLParser : L.OSM.JSONParser);
```

Reply:
Using `data instanceof XMLDocument` seems to work for me? Both with my suggested test parsing technique and debugging the current code with the data layer on. JSON is tricky because it just returns an `Object` rather than anything JSON specific...

Reply:
jquery isXMLDoc has a more complex logic, which at one point also checks `documentElement`: https://github.com/jquery/jquery/blob/098591e6fd3222e64b59af92c8849f5d8963d43c/src/core.js#L313-L320

Reply:
Ah it works with `DOMParser` in the browser but not with `jsdom` which returns something that seems to inherit from `Document` but which is a plain object?

Reply:
So if we add `jsdom-global` to the modules then this seems to work:

```
> require('jsdom-global')()
[Function: cleanup]
> parser = new window.DOMParser();
DOMParser {}
> data = parser.parseFromString("<osm><node/><node/></osm>", "text/xml");
Document { location: [Getter/Setter] }
> data instanceof Document
true
```

Unlike a true browser it returns `Document` not `XMLDocument` but `instanceof Document` works for both.

Reply:
Funny, I was just reading the exact same proposal here: https://stackoverflow.com/questions/42649700/using-domparser-in-javascript-testing-with-mocha-and-jsdom

I'll give it a try...

Reply:
Even though data was created from an XML file,  it's not an instance of the same Document that's being used for instanceof:

![image](https://github.com/user-attachments/assets/4dfdfc2f-f01f-4c5d-9950-9f6662c9b8ed)

Reply:
Ah, it was a missing `global.Document  = window.Document;`

I think it's working now.